src/platform_impl/macos/view.rs

```diff
@@ -433,22 +454,44 @@ extern "C" fn set_marked_text(
             marked_text.initWithString(string);
         };
         *marked_text_ref = marked_text;
+
+        let composed_string = marked_text_ref.clone().string();
+        let slice = slice::from_raw_parts(
+            composed_string.UTF8String() as *const c_uchar,
+            composed_string.len(),
+        );
+        let composed_string = str::from_utf8_unchecked(slice);
+
+        state.marked_text = composed_string.to_string();
```

Comment:
Now there's a `state.marked_text` as well as a `markedText` field. They seem to store the same text. I have a specific idea of resolving this. Would you mind if I'd added changes to this PR?

If you would not like me to add commits to this PR, that's completely okay, but then please remove either `state.marked_text` or `markedText`. Preferrably we would only have the `NSMutableAttributedString` stored within the `ViewState`.

Reply:
Welcome your patch !
I'm also confused there are duplicated information for `markedText` and `state.marked_text`.
`markedText` is referred in `setMarkedText` but I'm not sure how to remove this because I'm not familiar with Obj-c and type system between Obj-c and Rust.

I appreciate your help.